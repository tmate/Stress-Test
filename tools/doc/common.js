'use strict';

const yaml = require('js-yaml');

function isYAMLBlock(text) {
  return !!text.match(/^<!-- YAML/);
}

exports.isYAMLBlock = isYAMLBlock;

function arrify(value) {
  return Array.isArray(value) ? value : [value];
}

function extractAndParseYAML(text) {
  text = text.trim()
             .replace(/^<!-- YAML/, '')
             .replace(/-->$/, '');

  // js-yaml.safeLoad() throws on error
  const meta = yaml.safeLoad(text);

  if (meta.added) {
    // Since semver-minors can trickle down to previous major versions,
    // features may have been added in multiple versions.
    meta.added = arrify(meta.added);
  }

  if (meta.deprecated) {
    // Treat deprecated like added for consistency.
    meta.deprecated = arrify(meta.deprecated);
  }

  meta.changes = meta.changes || [];
  for (const entry of meta.changes) {
    entry.description = entry.description.replace(/^\^\s*/, '');
  }

  return meta;
}

exports.extractAndParseYAML = extractAndParseYAML;
