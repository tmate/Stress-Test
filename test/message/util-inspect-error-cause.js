'use strict';

require('../common');

const { inspect } = require('util');

class FoobarError extends Error {
  status = 'Feeling good';
}

const cause1 = new TypeError('Inner error');
const cause2 = new FoobarError('Individual message', { cause: cause1 });
cause2.extraProperties = 'Yes!';
const cause3 = new Error('Stack causes', { cause: cause2 });

process.nextTick(() => {
  const error = new RangeError('New Stack Frames', { cause: cause2 });
  const error2 = new RangeError('New Stack Frames', { cause: cause3 });

  inspect.defaultOptions.colors = true;

  console.log(inspect(error));
  console.log(inspect(cause3));
  console.log(inspect(error2));

  inspect.defaultOptions.colors = false;

  console.log(inspect(error));
  console.log(inspect(cause3));
  console.log(inspect(error2));
});
