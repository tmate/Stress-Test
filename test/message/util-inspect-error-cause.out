RangeError: New Stack Frames
    at *
*[90m    at *[39m {
  [cause]: FoobarError: Individual message
      at *
  *[90m    at *[39m
  *[90m    ... 4 lines matching cause stack trace ...*[39m
  *[90m    at *[39m {
    status: *[32m'Feeling good'*[39m,
    extraProperties: *[32m'Yes!'*[39m,
    [cause]: TypeError: Inner error
        at *
    *[90m    at *[39m
    *[90m    at *[39m
    *[90m    at *[39m
    *[90m    at *[39m
    *[90m    at *[39m
    *[90m    at *[39m
  }
}
Error: Stack causes
    at *
*[90m    at *[39m
*[90m    ... 4 lines matching cause stack trace ...*[39m
*[90m    at *[39m {
  [cause]: FoobarError: Individual message
      at *
  *[90m    at *[39m
  *[90m    ... 4 lines matching cause stack trace ...*[39m
  *[90m    at *[39m {
    status: *[32m'Feeling good'*[39m,
    extraProperties: *[32m'Yes!'*[39m,
    [cause]: TypeError: Inner error
        at *
    *[90m    at *[39m
    *[90m    at *[39m
    *[90m    at *[39m
    *[90m    at *[39m
    *[90m    at *[39m
    *[90m    at *[39m
  }
}
RangeError: New Stack Frames
    at *
*[90m    at *[39m {
  [cause]: Error: Stack causes
      at *
  *[90m    at *[39m
  *[90m    ... 4 lines matching cause stack trace ...*[39m
  *[90m    at *[39m {
    [cause]: FoobarError: Individual message
        at *
    *[90m    at *[39m
    *[90m    ... 4 lines matching cause stack trace ...*[39m
    *[90m    at *[39m {
      status: *[32m'Feeling good'*[39m,
      extraProperties: *[32m'Yes!'*[39m,
      [cause]: TypeError: Inner error
          at *
      *[90m    at *[39m
      *[90m    at *[39m
      *[90m    at *[39m
      *[90m    at *[39m
      *[90m    at *[39m
      *[90m    at *[39m
    }
  }
}
RangeError: New Stack Frames
    at *
    at * {
  [cause]: FoobarError: Individual message
      at *
      at *
      ... 4 lines matching cause stack trace ...
      at * {
    status: 'Feeling good',
    extraProperties: 'Yes!',
    [cause]: TypeError: Inner error
        at *
        at *
        at *
        at *
        at *
        at *
        at *
  }
}
Error: Stack causes
    at *
    at *
    ... 4 lines matching cause stack trace ...
    at * {
  [cause]: FoobarError: Individual message
      at *
      at *
      ... 4 lines matching cause stack trace ...
      at *
    status: 'Feeling good',
    extraProperties: 'Yes!',
    [cause]: TypeError: Inner error
        at *
        at *
        at *
        at *
        at *
        at *
        at *
  }
}
RangeError: New Stack Frames
    at *
    at * {
  [cause]: Error: Stack causes
      at *
      at *
      ... 4 lines matching cause stack trace ...
      at * {
    [cause]: FoobarError: Individual message
        at *
        at *
        ... 4 lines matching cause stack trace ...
        at * {
      status: 'Feeling good',
      extraProperties: 'Yes!',
      [cause]: TypeError: Inner error
          at *
          at *
          at *
          at *
          at *
          at *
          at *
    }
  }
}
